experimental/sql/src/impl_to_iterators.py

```diff
@@ -248,29 +248,27 @@ def match_and_rewrite(self, op: RelImpl.BinOp, rewriter: PatternRewriter):
 @dataclass
 class FullTableScanRewriter(RelImplRewriter):
 
+  # This table_mapping is assumed to be populated already. In this version, this
+  # is done through the use of get_batch_and_name_list. The batch types are then
+  # used to build BlockArguments and this dictionary is populated by mapping
+  # strings to the BlockArguments built from the corresponding types. Notice
+  # that we assume that the name of partial loads are in the form
+  # "table_name","col1","col2"....
   table_mapping: dict[str, it.ColumnarBatch]
 
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.FullTableScanOp,
                         rewriter: PatternRewriter):
-    rewriter.replace_matched_op(
-        it.ScanColumnarBatch.get(self.table_mapping[op.table_name.data],
-                                 convert_bag(op.result.typ)))
-
-
-@dataclass
-class PartialTableScanRewriter(RelImplRewriter):
-
-  table_mapping: dict[str, it.ColumnarBatch]
-
-  @op_type_rewrite_pattern
-  def match_and_rewrite(self, op: RelImpl.PartialTableScanOp,
-                        rewriter: PatternRewriter):
-    rewriter.replace_matched_op(
-        it.ScanColumnarBatch.get(
-            self.table_mapping[op.table_name.data + "," +
-                               ",".join([s.data for s in op.cols.data])],
-            convert_bag(op.result.typ)))
+    if op.cols:
+      rewriter.replace_matched_op(
+          it.ScanColumnarBatch.get(
+              self.table_mapping[op.table_name.data + "," +
+                                 ",".join([s.data for s in op.cols.data])],
+              convert_bag(op.result.typ)))
+    else:
+      rewriter.replace_matched_op(
+          it.ScanColumnarBatch.get(self.table_mapping[op.table_name.data],
+                                   convert_bag(op.result.typ)))
```

Comment:
Maybe you can factor out `table_name` (or similar), which is the only difference between the two cases.